### PR TITLE
Update simple_functions.py

### PR DESCRIPTION
**Pull request by mfouesneau**
Doc and types

### CODE REVIEW

File: participants/Fouesneau/pr_tutorial/simple_functions.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+def fibonacci(max: int) -> List[int]:
+    """ Calculate the Fibonacci series up to max """
     values = [0, 1]
     while values[-2] + values[-1] < max:
         values.append(values[-2] + values[-1])
     return values
 
 
-def factorial(value):
+def factorial(value: int) -> int:
```
Comment (cphyc) [inline participants/Fouesneau/pr_tutorial/simple_functions.py:11]: Just realized that the function will only work if the value is positive. Maybe it'd be great to add this in the docstring?

**Comment by mfouesneau** [comment]
Good point. Made changes in the doc and exception raised

File: participants/Fouesneau/pr_tutorial/simple_functions.py (hunk 1 above)
Comment (cphyc) [reply participants/Fouesneau/pr_tutorial/simple_functions.py:11]: And also would it be worth preventing the user with a positional argument like `factorial(value=something)`? We could use the following
```suggestion
def factorial(value: int, /) -> int:
```

Comment (mfouesneau) [reply participants/Fouesneau/pr_tutorial/simple_functions.py:11]: Thanks, but that's a style change. The example you mention will raise an error already

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Python package
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    working-directory: 2022_contributing_to_open_source/participants/Fouesneau/
```
Comment (cphyc) [inline .github/workflows/ci.yml]: On my local machine, I get an error while trying to run this workflow. Do you know of an online tool to validate Github Action yaml files?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: Python package
+
+on: [push, pull_request]
```
Comment (cphyc) [inline .github/workflows/ci.yml]: I am unsure about the syntax, but I think this should be

```suggestion
on:
    push:
    pull_request:
```

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mfouesneau) [reply .github/workflows/ci.yml]: Yes, this is not meant to be committed. Mistake.

File: participants/Fouesneau/pr_tutorial/simple_functions.py [hunk 4]
```diff
@@ -1,11 +1,17 @@
-def fibonacci(max):
+from typing import Sequence
+
+def fibonacci(max: int) -> List[int]:
```
Comment (cphyc) [inline participants/Fouesneau/pr_tutorial/simple_functions.py:3]: I think `List` is missing its import and `Sequence` is unused. I'd suggest
```suggestion
from typing import List

def fibonacci(max: int) -> List[int]:
```

File: participants/Fouesneau/pr_tutorial/simple_functions.py (hunk 1 above)
Comment (cphyc) [reply participants/Fouesneau/pr_tutorial/simple_functions.py:11]: On my machine, it seems to be working ok
```python
In [4]: factorial(value=10)
Out[4]: 3628800
```
But that's indeed purely stylistic :)

File: participants/Fouesneau/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+requires = ["setuptools",
+            "setuptools-scm",
+            "wheel",
+            "extension-helpers"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "ahw2022-pr-mf"
+description = "Exercise from AWH2022"
+version = "0.1.0"
```
Comment (cphyc) [inline participants/Fouesneau/pyproject.toml:17]: Should we strip the `setup.py` file from its content since it is all replicated here?

**Comment by ceb8** [comment]
Thanks! 😄

**Pull request merged**